packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationModule.m

```diff
@@ -0,0 +1,516 @@
+// Copyright 2019-present 650 Industries. All rights reserved.
+
+#import <EXScreenOrientation/EXScreenOrientationModule.h>
+#import <UMCore/UMEventEmitterService.h>
+#import <EXScreenOrientation/EXScreenOrientationRegistry.h>
+#import <UMCore/UMModuleRegistryProvider.h>
+
+#import <UIKit/UIKit.h>
+#import <sys/utsname.h>
+
+@interface EXScreenOrientationModule ()
+
+@property (nonatomic, weak) UMModuleRegistry *moduleRegistry;
+@property (nonatomic, weak) id<UMEventEmitterService> eventEmitter;
+@property (nonatomic) bool hasListeners;
+@property (nonatomic) UIInterfaceOrientation currentScreenOrientation;
+
+@end
+
+static int INVALID_MASK = 0;
+static NSString *defaultAppId = @"bareAppId";
+
+@implementation EXScreenOrientationModule
+
+UM_EXPORT_MODULE(ExpoScreenOrientation);
+
+- (void)setModuleRegistry:(UMModuleRegistry *)moduleRegistry
+{
+  _moduleRegistry = moduleRegistry;
+  _eventEmitter = [moduleRegistry getModuleImplementingProtocol:@protocol(UMEventEmitterService)];
+  _hasListeners = NO;
+  UIDevice *device = [UIDevice currentDevice];
+  UIInterfaceOrientation currentDeviceOrientation = [self UIDeviceOrientationToUIInterfaceOrientation:[device orientation]];
+  UIInterfaceOrientationMask orientationMask = [self orientationMask];
+  
+  // this gives the correct information of screen orientation before any rotation or locking
+  if ([self doesOrientationMask:orientationMask containOrientation:currentDeviceOrientation]) {
+    _currentScreenOrientation = currentDeviceOrientation;
+  }
+  else {
+    _currentScreenOrientation = [self defaultOrientationForOrientationMask:[self orientationMask]];
+  }
+  [[UIDevice currentDevice] beginGeneratingDeviceOrientationNotifications];
+  [[NSNotificationCenter defaultCenter] addObserver:self
```

Comment:
We don't ever removeObserver for this file and it is persisted strongly, so we never remove references to these classes

Reply:
I added this to init method and removeObserver to dealloc. Is it ok?